src/@context/Web3.tsx

```diff
@@ -155,16 +157,33 @@ function Web3Provider({ children }: { children: ReactNode }): ReactElement {
     if (!accountId || !networkId || !web3) return
 
     try {
+      const { oceanTokenAddress } = getOceanConfig(networkId)
+      const approvedTokensBalance: { [key: string]: string } = {}
+      const approvedTokens = (await getOpcsApprovedTokens(chainId)).filter(
+        (token) => token.address !== oceanTokenAddress
+      )
+      approvedTokens.forEach(async (token) => {
+        const { address, decimals, symbol } = token
+        const tokenBalance = await getTokenBalance(
+          accountId,
+          decimals,
+          address,
+          web3
+        )
+        approvedTokensBalance[symbol.toLocaleLowerCase()] = tokenBalance
+      })
+
       const balance = {
         eth: web3.utils.fromWei(await web3.eth.getBalance(accountId, 'latest')),
-        ocean: await getOceanBalance(accountId, networkId, web3)
+        ocean: await getOceanBalance(accountId, networkId, web3),
```

Comment:
bonus points for not making duplication in codebase worse: `getOceanBalance()` from `@utils/ocean` is redundant and unnecessarily specific, so can be replaced completely with the newly created `getTokenBalance()`.

Reply:
I refactored the user balance fetch and took the occasion to move the approvedBaseTokens list into the web3 provider to reduce the number of calls of the `getOpcsApprovedTokens()` function.